src/vs/workbench/contrib/tasks/browser/taskQuickPick.ts

```diff
@@ -265,6 +291,74 @@ export class TaskQuickPick extends Disposable {
 		return;
 	}
 
+	private async _setColor(task: Task | ConfiguringTask | null | string | undefined): Promise<void> {
+		if (task === undefined || task === null || typeof task === 'string') {
+			return;
+		}
+		const colorTheme = this._themeService.getColorTheme();
+		const standardColors: string[] = getStandardColors(colorTheme);
+		const styleElement = getColorStyleElement(colorTheme);
+		const items: (IQuickPickItem | IQuickPickSeparator)[] = [];
+		for (const colorKey of standardColors) {
+			const colorClass = getColorClass(colorKey);
+			items.push({
+				label: `$(${Codicon.circleFilled.id}) ${colorKey.replace('terminal.ansi', '')}`, id: colorKey, description: colorKey, iconClasses: [colorClass]
+			});
+		}
+		items.push({ type: 'separator' });
+		const showAllColorsItem = { label: 'Reset to default' };
+		items.push(showAllColorsItem);
+		document.body.appendChild(styleElement);
+
+		const quickPick = this._quickInputService.createQuickPick();
+		quickPick.items = items;
+		quickPick.matchOnDescription = true;
+		quickPick.show();
+		const disposables: IDisposable[] = [];
+		const result = await new Promise<IQuickPickItem | undefined>(r => {
+			disposables.push(quickPick.onDidHide(() => r(undefined)));
+			disposables.push(quickPick.onDidAccept(() => r(quickPick.selectedItems[0])));
+		});
+		dispose(disposables);
+
+		if (result && task && typeof task !== 'string') {
+			task.configurationProperties.color = result.id;
+		}
+		document.body.removeChild(styleElement);
+		quickPick.hide();
```

Comment:
I see in `_setIcon` you're calling `this._taskService.customize` to set the icon, which makes sense. I don't see that in `_setColor` though.

Reply:
color gets set in `_setIcon`